src/main/java/roboy/dialog/tutorials/StateMachineExamples.java

```diff
@@ -51,20 +53,21 @@ public static void main(String[] args) throws Exception {
 
     private static DialogStateMachine fromCode() {
 
-
+        // 0. Create an InferenceEngine
+        InferenceEngine inference = new Inference();
```

Comment:
if we pass inference here, why don't we pass ros node and memory as well?
previously we omitted ros node and memory to show a minimal usage example

Reply:
inference will be always a part of the dialog, no? whereas memory and ros node can be omitted

Reply:
hmm makes sense
but then we should change the order of arguments so that inference engine becomes the first argument 
new StateMachine(null, null, null, inference) looks weird